### ISSUE TITLE
Strange error after update from 1.37.20 to 1.38.0

### ISSUE THREAD

**Issue by newnon** [kripken/emscripten#6519]
i've got strange error after update when i try to build asmjs version wasm version compile successful  

Traceback (most recent call last):
  File "/Development/emsdk-master/emscripten/1.38.0/emcc", line 11, in <module>
    python_selector.run(__file__)
  File "/Development/emsdk-master/emscripten/1.38.0/tools/python_selector.py", line 38, in run
    sys.exit(run_by_import(filename, main) if on_allowed_version() else run_by_subprocess(filename))
  File "/Development/emsdk-master/emscripten/1.38.0/tools/python_selector.py", line 13, in run_by_import
    return getattr(importlib.import_module(os.path.basename(filename)), main)()
  File "/Development/emsdk-master/emscripten/1.38.0/emcc.py", line 1864, in run
    optimizer.flush()
  File "/Development/emsdk-master/emscripten/1.38.0/emcc.py", line 241, in flush
    self.run_passes(chunks[0], title, just_split=False, just_concat=False)
  File "/Development/emsdk-master/emscripten/1.38.0/emcc.py", line 276, in run_passes
    output_filename=self.in_temp(os.path.basename(final) + '.jsopted.js'))
  File "/Development/emsdk-master/emscripten/1.38.0/tools/shared.py", line 2086, in js_optimizer
    ret = js_optimizer.run(filename, passes, NODE_JS, debug, extra_info, just_split, just_concat)
  File "/Development/emsdk-master/emscripten/1.38.0/tools/js_optimizer.py", line 557, in run
    return temp_files.run_and_clean(lambda: run_on_js(filename, passes, js_engine, source_map, extra_info, just_split, just_concat))
  File "/Development/emsdk-master/emscripten/1.38.0/tools/tempfiles.py", line 93, in run_and_clean
    return func()
  File "/Development/emsdk-master/emscripten/1.38.0/tools/js_optimizer.py", line 557, in <lambda>
    return temp_files.run_and_clean(lambda: run_on_js(filename, passes, js_engine, source_map, extra_info, just_split, just_concat))
  File "/Development/emsdk-master/emscripten/1.38.0/tools/js_optimizer.py", line 492, in run_on_js
    cld = shared.Building.closure_compiler(cld, pretty='minifyWhitespace' not in passes)
  File "/Development/emsdk-master/emscripten/1.38.0/tools/shared.py", line 2218, in closure_compiler
    raise Exception('closure compiler error: ' + process.stdout + ' (rc: %d)' % process.returncode)
Exception: closure compiler error: /Development/emsdk-master/emscripten/1.38.0/third_party/closure-compiler/node-externs/domain.js:67: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @param {function(...[*])} callback
                       ^

/Development/emsdk-master/emscripten/1.38.0/third_party/closure-compiler/node-externs/domain.js:68: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @return {function(...[*])}
                        ^

/Development/emsdk-master/emscripten/1.38.0/third_party/closure-compiler/node-externs/domain.js:73: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @param {function(...[*])} callback
                       ^

/Development/emsdk-master/emscripten/1.38.0/third_party/closure-compiler/node-externs/domain.js:74: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.
 * @return {function(...[*])}
                        ^

/tmp/tmpARZIIs.cl.js:2841: ERROR - variable onmessage is undeclared
        onmessage(message);
        ^^^^^^^^^

**Comment by kripken** [comment]
It might be worth looking in the versions in the middle, to see where this started to break.

But, it looks like a closure compiler error - we upgraded closure at some point, and it has better/stronger warnings and errors. We should find out if the error is from emscripten code or JS code that you added - if you can submit a testcase, I can look at that.

If it's your own code and you can't find a workaround, the `IGNORE_CLOSURE_COMPILER_ERRORS` might help.

**Comment by newnon** [comment]
unfortunately i have huge project and at this point i don't know wats part did break compilation process but i will try  IGNORE_CLOSURE_COMPILER_ERRORS

**Comment by newnon** [comment]
ok finally i've got test case
```
worker.cpp
#include <emscripten/emscripten.h>

extern "C" {

void getWinRateInBackGround(char* data, int size) {
    float ret = 0.0f;
    emscripten_worker_respond((char*)&ret, sizeof(ret));
}
}
```
build command

${CMAKE_C_COMPILER} ${CMAKE_CURRENT_SOURCE_DIR}/worker.cpp -std=c++11 --memory-init-file 0 -s EXPORTED_FUNCTIONS="['_getWinRateInBackGround']" -s BUILD_AS_WORKER=1 -o worker.js --llvm-lto 1 -Os --closure 1  -s NO_EXIT_RUNTIME=1 -s DEMANGLE_SUPPORT=0 -s ELIMINATE_DUPLICATE_FUNCTIONS=1 -s NO_FILESYSTEM=1

**Comment by kripken** [comment]
Thanks for the testcase, should be fixed by #6527
